### PR TITLE
EVSE - Update FakeReadings when a MaxChargeCurrent change occurs

### PR DESCRIPTION
**Pull request by jamesharrow**
This PR (not required for SVE) helps make the EVSE fake readings follow changes to the maxChargeCurrent.

- To use it, send a TestEventTrigger to start the Fake Load test
- Once this is running it will default to 1000W (+/-20W)
- Thereafter changes to the MaxChargeCurrent (e.g. from setting the UserMaximumChargeCurrent) will recalculate power_mW and update the current_mA in the fake readings. This can be useful for demoing representative EV charging loads without a real EV connected.

### CODE REVIEW

**Review by PeterC1965 — APPROVED** [review]

File: examples/energy-management-app/energy-management-common/energy-evse/include/EVSEManufacturerImpl.h [hunk 1]
```diff
@@ -213,6 +213,13 @@ class EVSEManufacturer : public DEMManufacturerDelegate
      */
     static void FakeReadingsTimerExpiry(System::Layer * systemLayer, void * manufacturer);
 
+    /*
+     * @brief   Updates the parameters used to generate fake power and energy readings
+     *
+     * @param   Amperage_mA   Maximum Charge current in mA
```
Comment (andy31415) [inline examples/energy-management-app/energy-management-common/energy-evse/include/EVSEManufacturerImpl.h:219]:
```suggestion
     * @param   maximumChargeCurrent   Maximum Charge current in mA
```

File: examples/energy-management-app/energy-management-common/energy-evse/src/EVSEManufacturerImpl.cpp [hunk 2]
```diff
@@ -551,6 +552,12 @@ CHIP_ERROR EVSEManufacturer::SendPeriodicEnergyReading(EndpointId aEndpointId, i
     return CHIP_NO_ERROR;
 }
 
+void EVSEManufacturer::UpdateEVFakeReadings(const Amperage_mA maximumChargeCurrent)
+{
+    FakeReadings::GetInstance().SetCurrent(maximumChargeCurrent);
+    FakeReadings::GetInstance().SetPower((FakeReadings::GetInstance().GetVoltage() * maximumChargeCurrent) / 1000);
```
Comment (andy31415) [inline examples/energy-management-app/energy-management-common/energy-evse/src/EVSEManufacturerImpl.cpp:558]: could you add a comment on why the division by 1000 ?  I assume this is to make the mA*mV be mW

File: examples/energy-management-app/energy-management-common/energy-reporting/src/FakeReadings.cpp [hunk 3]
```diff
@@ -180,3 +180,60 @@ void FakeReadings::FakeReadingsTimerExpiry(System::Layer * systemLayer, void * m
 
     mn->FakeReadingsUpdate();
 }
+
+/**
+ * @brief   Sets the fake Power level to use
```
Comment (andy31415) [inline examples/energy-management-app/energy-management-common/energy-reporting/src/FakeReadings.cpp:185]: nit: these comments seem to just waste vertical space. It is super obvious that a call of `void SetFoo(foo_unit value)` "sets the value of Foo in unit units".

**Review by andy31415 — APPROVED** [review]

File: examples/energy-management-app/energy-management-common/energy-evse/src/EVSEManufacturerImpl.cpp (hunk 2 above)
Comment (jamesharrow) [inline examples/energy-management-app/energy-management-common/energy-evse/src/EVSEManufacturerImpl.cpp:558]:
```suggestion
    // Note we have to divide by 1000 to make ma * mv = mW
    FakeReadings::GetInstance().SetPower((FakeReadings::GetInstance().GetVoltage() * maximumChargeCurrent) / 1000);
```

File: examples/energy-management-app/energy-management-common/energy-reporting/src/FakeReadings.cpp [hunk 4]
```diff
@@ -180,3 +180,60 @@ void FakeReadings::FakeReadingsTimerExpiry(System::Layer * systemLayer, void * m
...
+ */
+void FakeReadings::SetPower(Power_mW aPower_mW)
+{
+    mPower_mW = aPower_mW;
+}
+/**
+ * @brief   Gets the fake Power level in use
+ *
+ * @return  Power in mW
+ */
```
Comment (jamesharrow) [inline examples/energy-management-app/energy-management-common/energy-reporting/src/FakeReadings.cpp:197]:
```suggestion
```

File: examples/energy-management-app/energy-management-common/energy-reporting/src/FakeReadings.cpp [hunk 5]
```diff
@@ -180,3 +180,60 @@ void FakeReadings::FakeReadingsTimerExpiry(System::Layer * systemLayer, void * m
...
+Power_mW FakeReadings::GetPower()
+{
+    return mPower_mW;
+};
+/**
+ * @brief   Sets the fake Voltage level to use
+ *
+ * @param   Voltage in mV
+ */
```
Comment (jamesharrow) [inline examples/energy-management-app/energy-management-common/energy-reporting/src/FakeReadings.cpp:206]:
```suggestion

```

File: examples/energy-management-app/energy-management-common/energy-reporting/src/FakeReadings.cpp [hunk 6]
```diff
@@ -180,3 +180,60 @@ void FakeReadings::FakeReadingsTimerExpiry(System::Layer * systemLayer, void * m
...
+void FakeReadings::SetVoltage(Voltage_mV aVoltage_mV)
+{
+    mVoltage_mV = aVoltage_mV;
+}
+/**
+ * @brief   Gets the fake Voltage level in use
+ *
+ * @return  Voltage in mV
+ */
```
Comment (jamesharrow) [inline examples/energy-management-app/energy-management-common/energy-reporting/src/FakeReadings.cpp:215]:
```suggestion
```

File: examples/energy-management-app/energy-management-common/energy-reporting/src/FakeReadings.cpp [hunk 7]
```diff
@@ -180,3 +180,60 @@ void FakeReadings::FakeReadingsTimerExpiry(System::Layer * systemLayer, void * m
...
+Voltage_mV FakeReadings::GetVoltage()
+{
+    return mVoltage_mV;
+};
+
+/**
+ * @brief   Sets the fake Current level to use
+ *
+ * @param   Current in mA
+ */
```
Comment (jamesharrow) [inline examples/energy-management-app/energy-management-common/energy-reporting/src/FakeReadings.cpp:225]:
```suggestion

```

File: examples/energy-management-app/energy-management-common/energy-reporting/src/FakeReadings.cpp [hunk 8]
```diff
@@ -180,3 +180,60 @@ void FakeReadings::FakeReadingsTimerExpiry(System::Layer * systemLayer, void * m
...
+void FakeReadings::SetCurrent(Amperage_mA aCurrent_mA)
+{
+    mCurrent_mA = aCurrent_mA;
+}
+
+/**
+ * @brief   Gets the fake Current level in use
+ *
+ * @return  Current in mA
+ */
```
Comment (jamesharrow) [inline examples/energy-management-app/energy-management-common/energy-reporting/src/FakeReadings.cpp:235]:
```suggestion

```

File: examples/energy-management-app/energy-management-common/energy-reporting/src/FakeReadings.cpp [hunk 9]
```diff
@@ -180,3 +180,60 @@ void FakeReadings::FakeReadingsTimerExpiry(System::Layer * systemLayer, void * m
 
     mn->FakeReadingsUpdate();
 }
+
+/**
+ * @brief   Sets the fake Power level to use
+ *
+ * @param   Power in mW
+ */
```
Comment (jamesharrow) [inline examples/energy-management-app/energy-management-common/energy-reporting/src/FakeReadings.cpp:188]:
```suggestion


```

File: examples/energy-management-app/energy-management-common/energy-reporting/src/FakeReadings.cpp (hunk 3 above)
Comment (jamesharrow) [reply examples/energy-management-app/energy-management-common/energy-reporting/src/FakeReadings.cpp:185]: Agree - ok removing them.

File: examples/energy-management-app/energy-management-common/energy-reporting/src/FakeReadings.cpp [hunk 10]
```diff
@@ -180,3 +180,31 @@ void FakeReadings::FakeReadingsTimerExpiry(System::Layer * systemLayer, void * m
...
+Voltage_mV FakeReadings::GetVoltage()
+{
+    return mVoltage_mV;
+};
+
+void FakeReadings::SetCurrent(Amperage_mA aCurrent_mA)
+{
+    mCurrent_mA = aCurrent_mA;
+}
+
```
Comment (jamesharrow) [inline examples/energy-management-app/energy-management-common/energy-reporting/src/FakeReadings.cpp:206]:
```suggestion
```

File: examples/energy-management-app/energy-management-common/energy-reporting/src/FakeReadings.cpp [hunk 11]
```diff
@@ -180,3 +180,31 @@ void FakeReadings::FakeReadingsTimerExpiry(System::Layer * systemLayer, void * m
...
+};
+void FakeReadings::SetVoltage(Voltage_mV aVoltage_mV)
+{
+    mVoltage_mV = aVoltage_mV;
+}
+Voltage_mV FakeReadings::GetVoltage()
+{
+    return mVoltage_mV;
+};
+
```
Comment (jamesharrow) [inline examples/energy-management-app/energy-management-common/energy-reporting/src/FakeReadings.cpp:201]:
```suggestion
```

**Pull request merged**